app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -30,12 +31,19 @@ def connect(options = {})
     # Create the underlying connection according to the version of the oVirt API requested by
     # the caller:
     connect_method = "raw_connect_v#{version}".to_sym
-    connection = self.class.public_send(connect_method, server, port, path, username, password, service)
+    unless @connection.nil?
```

Comment:
What happens if the connection is closed? it will not be nil but this will return a closed connection.